model_interfaces/ltm_agent_1.py

```diff
@@ -313,6 +318,32 @@ def reset_all(self):
     def reset(self):
         self.reset_all()
 
+    def save(self):
+        infos = [self.message_history, self.user_info, self.text_mem.state_as_text()]
+        files = ["_message_hist.json", "_user_info.json", "_mem.json"]
+
+        for obj, file_ext in zip(infos, files):
+            fname = PERSISTENCE_DIR.joinpath(self.save_name + file_ext)
+            with open(fname, "w") as fd:
+                json.dump(obj, fd, cls=CustomEncoder)
+
+    def load(self):
+        fname = PERSISTENCE_DIR.joinpath(self.save_name + "_message_hist.json")
+        with open(fname, "r") as fd:
+            ctx = json.load(fd)
+
+        message_hist = []
+        for m in ctx:
+            message_hist.append(Message(**m))
+        self.message_history = message_hist
+
+        fname = PERSISTENCE_DIR.joinpath(self.save_name + "_user_info.json")
+        with open(fname, "r") as fd:
+            self.user_info = json.load(fd)
+
+        fname = PERSISTENCE_DIR.joinpath(self.save_name + "_mem.json")
+        with open(fname, "r") as fd:
+            self.text_mem.set_state(json.load(fd))
```

Comment:
Since the base class already provides `self.save_path` which is a directory, we could save those files in there e.g.
```python
fname = self.save_path.joinpath("mem.json")
```